docs/source/hooks.rst

```diff
@@ -40,13 +40,15 @@ Training loop
 ^^^^^^^^^^^^^
 
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_epoch_start`
+- :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_train_batch_start`
 - :meth:`~pytorch_lightning.core.lightning.LightningModule.tbptt_split_batch`
 - :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step`
 - :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step_end` (optional)
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_before_zero_grad`
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.backward`
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_after_backward`
 - ``optimizer.step()``
+- :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_train_batch_end`
 - :meth:`~pytorch_lightning.core.lightning.LightningModule.training_epoch_end`
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_epoch_end`
```

Comment:
Why keep `on_epoch_end` if we have `on_train_epoch_end`? They are basically doing the same thing right? Same with `on_epoch_start`.

Reply:
true. mind removing those?

@Borda

Reply:
yeah sure.

Reply:
I see no deprecation has been set in the docstrings yet for these ModelHooks, should I still remove them? Also, similar hooks are present in Callback as well `on_batch_start`, `on_batch_end`, `on_epoch_start`, `on_epoch_end`. Should we remove them all since all of them have their alternatives with prefix `on_train_*`??

@Borda